src/Code/DefinitionSummaryTooltip.elm

```diff
@@ -29,48 +30,73 @@ type DefinitionSummary
     | DataConstructorHover DataConstructorSummary
     | AbilityConstructorHover AbilityConstructorSummary
 
-
 type alias Model =
-    Maybe ( Reference, WebData DefinitionSummary )
+    {
+        data: Maybe ( Reference, WebData DefinitionSummary )
+        , alreadyFetched: Dict String (Reference, WebData DefinitionSummary)
+    }
 
 
 init : Model
 init =
-    Nothing
+    {
+        data = Nothing
+        , alreadyFetched = Dict.empty
+    }
 
 
 
 -- UPDATE
 
 
 type Msg
-    = ShowTooltip Reference
+    = ShowTooltipDelayed Reference
+    | ShowTooltip Reference
     | HideTooltip Reference
+    | CachedDefinition Reference (WebData DefinitionSummary)
     | FetchDefinitionFinished Reference (HttpResult DefinitionSummary)
 
-
 update : Config -> Msg -> Model -> ( Model, Cmd Msg )
 update config msg model =
+    let debounceDelay = 500
```

Comment:
How does it feel with 300?

Reply:
I actually first tried 300 as it is used some other place in the code. But than bumped it up to 2000 when testing and slowly got down to 500 for some reason forgetting about the initial 300 🙈
But yeah, 300 feels snappier :)